File: learnAPI/1b.py
import requests
import pandas as pd

url = "https://api.coindesk.com/v1/bpi/currentprice.json"

response = requests.get(url)

if response.status_code == 200:
    data = response.json()
    
    df = pd.DataFrame({
        'USD': data['bpi']['USD'],
        'GBP': data['bpi']['GBP'],
        'EUR': data['bpi']['EUR']
    })
    print(df)
    
    
    for key, items in data['time'].items():
        print('Type: ', key, ',   ', 'Time: ', items)
    
    
else:
    print("Failed to retrieve data:", response.status_code)

File: learnAPI/1b.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

url <- "https://api.coindesk.com/v1/bpi/currentprice.json"

response <- GET(url)

if (http_status(response)$category == "Success") {
    json_content <- content(response, "text")
    data <- fromJSON(json_content)

    df <- t(as.data.frame(data$bpi$USD))
    colnames(df) <- 'USD'
    print(df)

    df <- t(as.data.frame(data$time))
    colnames(df) <-'time'
    print(df)


    for (key in names(data$time)){
        print(key)
    }
    for (key in data$time){
        print(key)
    }
} else {
    cat("Permintaan tidak berhasil. Kode status:", http_status(response)$status_code)
}

